src/connect.rs

```diff
@@ -10,6 +10,30 @@ use tungstenite::{
 use crate::{domain, stream::MaybeTlsStream, Connector, IntoClientRequest, WebSocketStream};
 
 /// Connect to a given URL.
+///
+/// Accepts any request that implements [`IntoClientRequest`], which is often just `&str`, but can
+/// be a variety of types such as `httparse::Request` or [`tungstenite::http::Request`] for more
+/// complex uses.
+///
+/// ```no_run
+/// # async fn example() {
+/// use tungstenite::http::{Method, Request};
+/// use tokio_tungstenite::connect_async;
+///
+/// let request = Request::builder()
+///             .uri("wss://api.example.com")
+///             .method(Method::GET)
+///             .header("Sec-WebSocket-Key", "someUniqueValue")
+///             .header("Sec-WebSocket-Version", "13")
+///             .header("host", "api.example.com")
+///             .header("Connection", "Upgrade")
+///             .header("Upgrade", "websocket")
+///             .body(())
+///             .unwrap();
```

Comment:
I think this kind of pattern is actually a bit problematic as it requires specifying all WebSocket headers properly. A more typical issue that people experience is that they just want to add one or more additional headers on top of what we generate automatically, see https://github.com/snapview/tungstenite-rs/issues/327

Reply:
Ah, that's even better for my use case, too. I've updated the example if you're okay with adding it. Thanks!